### DISCUSSION TITLE
How can i hide my real ip

### DISCUSSION THREAD

**Discussion by ggforces** [seleniumbase/SeleniumBase#2276] [Q&A] [answered]
Pixelscan.net detects that I'm using a proxy and can see my real IP address. How can I prevent this?

```
from seleniumbase import SB
proxy = "un:pw@ip:port"
with SB(uc=True,locale_code="EN", headless=False,undetectable=True,agent="Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36",proxy=proxy) as sb:

        try:
            sb.get("about:blank")
            while True:
                sb.execute_script("return navigator.userAgent;")
        except:
            pass
print("Finished")
```
![image](https://github.com/seleniumbase/SeleniumBase/assets/60465269/4e18d46a-5782-4951-ab4b-a4c268440daa)

**Comment by mdmintz** [comment, marked as answer]
As mentioned here:
https://github.com/ultrafunkamsterdam/undetected-chromedriver/blob/cea80717c5a3d95ccf5c40e6e38081d5454ec7a5/README.md?plain=1#L35

<img width="930" alt="Screenshot 2023-11-13 at 9 56 56 PM" src="https://github.com/seleniumbase/SeleniumBase/assets/6788579/3089014b-11a0-42da-931a-ac0f28cefbde">

As you saw, it says: "No automation framework detected", which means UC Mode did its job. It makes your bot look human.

It does not hide your IP Address. You can change your proxy settings to change your IP, but they may be able to detect that.

**Reply by ggforces** [reply]
What I was trying to convey was the method of changing the proxy. For example, when using selenium-wire with undetected chromedriver v2, it was detectable. However, this was not the case when used in the non-v2 version. I believe a similar situation exists in the UC mode of SeleniumBase. When proxy information is directly entered, the actual IP address is exposed. The issue here is not whether the proxy is realistic but rather the revelation of the real IP address due to the method of using the proxy. Do you have any other method to suggest for changing the IP integrated into SeleniumBase, excluding selenium-wire?

**Reply by mdmintz** [reply]
That `proxy` arg is the only way of changing the proxy in UC Mode. The command-line option, `--proxy=PROXY` is another way of instantiating it.

If not using UC Mode, then changing the proxy can also be done in Wire Mode. But since the UC Mode option is a higher priority for you, then the Wire Mode way isn't a valid option.
